packages/block-library/src/gallery/gallery-image.js

```diff
@@ -160,15 +160,17 @@ class GalleryImage extends Component {
 						disabled={ ! isSelected }
 					/>
 				</div>
-				<RichText
-					tagName="figcaption"
-					placeholder={ isSelected ? __( 'Write caption…' ) : null }
-					value={ caption }
-					isSelected={ this.state.captionSelected }
-					onChange={ ( newCaption ) => setAttributes( { caption: newCaption } ) }
-					unstableOnFocus={ this.onSelectCaption }
-					inlineToolbar
-				/>
+				{ ( isSelected || caption ) && (
+					<RichText
+						tagName="figcaption"
+						placeholder={ isSelected ? __( 'Write caption…' ) : null }
```

Comment:
the check here is useless now right?

Reply:
Because the placeholder won't show if there is a caption?